pkg/clients/database/sql.go

```diff
@@ -0,0 +1,612 @@
+		Size:     skuSpec.Size,
+	}, nil
+}
+
+// ToPostgreSQLSKU returns the name of the PostgreSQL SKU, which is tier + family + cores, e.g. B_Gen4_1, GP_Gen5_8.
+func ToPostgreSQLSKU(skuSpec azuredbv1alpha3.SKU) (*postgresql.Sku, error) {
+	t, ok := skuShortTiers[mysql.SkuTier(skuSpec.Tier)]
+	if !ok {
+		return nil, fmt.Errorf("tier '%s' is not one of the supported values: %+v", skuSpec.Tier, mysql.PossibleSkuTierValues())
+	}
+	return &postgresql.Sku{
+		Name:     azure.ToStringPtr(fmt.Sprintf("%s_%s_%s", t, skuSpec.Family, strconv.Itoa(skuSpec.Capacity))),
+		Tier:     postgresql.SkuTier(skuSpec.Tier),
+		Capacity: azure.ToInt32Ptr(skuSpec.Capacity),
+		Family:   azure.ToStringPtr(skuSpec.Family),
+		Size:     skuSpec.Size,
+	}, nil
+}
+
+// GeneratePostgreSQLObservation produces SQLServerObservation from postgresql.Server.
+func GeneratePostgreSQLObservation(in postgresql.Server) azuredbv1alpha3.SQLServerObservation {
+	return azuredbv1alpha3.SQLServerObservation{
+		ID:                       azure.ToString(in.ID),
+		Name:                     azure.ToString(in.Name),
+		Type:                     azure.ToString(in.Type),
+		UserVisibleState:         string(in.UserVisibleState),
+		FullyQualifiedDomainName: azure.ToString(in.FullyQualifiedDomainName),
+		MasterServerID:           azure.ToString(in.MasterServerID),
+	}
+}
+
+// GenerateMySQLObservation produces SQLServerObservation from mysql.Server
+func GenerateMySQLObservation(in mysql.Server) azuredbv1alpha3.SQLServerObservation {
+	return azuredbv1alpha3.SQLServerObservation{
+		ID:                       azure.ToString(in.ID),
+		Name:                     azure.ToString(in.Name),
+		Type:                     azure.ToString(in.Type),
+		UserVisibleState:         string(in.UserVisibleState),
+		FullyQualifiedDomainName: azure.ToString(in.FullyQualifiedDomainName),
+		MasterServerID:           azure.ToString(in.MasterServerID),
+	}
+}
+
+// LateInitializeMySQL fills the empty values of SQLServerParameters with the
+// ones that are retrieved from the Azure API.
+func LateInitializeMySQL(p *azuredbv1alpha3.SQLServerParameters, in mysql.Server) {
+	if in.Sku != nil {
+		p.SKU.Size = azure.LateInitializeStringPtrFromPtr(p.SKU.Size, in.Sku.Size)
+	}
+	p.Tags = azure.LateInitializeStringMap(p.Tags, in.Tags)
+	if in.StorageProfile != nil {
+		p.StorageProfile.BackupRetentionDays = azure.LateInitializeIntPtrFromInt32Ptr(p.StorageProfile.BackupRetentionDays, in.StorageProfile.BackupRetentionDays)
+		p.StorageProfile.GeoRedundantBackup = azure.LateInitializeStringPtrFromVal(p.StorageProfile.GeoRedundantBackup, string(in.StorageProfile.GeoRedundantBackup))
+		p.StorageProfile.StorageAutogrow = azure.LateInitializeStringPtrFromVal(p.StorageProfile.StorageAutogrow, string(in.StorageProfile.StorageAutogrow))
+	}
+}
+
+// IsMySQLUpToDate is used to report whether given mysql.Server is in
+// sync with the SQLServerParameters that user desires.
+// nolint:gocyclo
```

Comment:
Nitpick: my preference is to put these `nolint` directives right after the opening brace of the function, so they don't leak into the GoDoc.